src/app/component/circular-heatmap/circular-heatmap.component.ts

```diff
@@ -58,124 +65,116 @@ export class CircularHeatmapComponent implements OnInit {
-    this.yaml.setURI('./assets/YAML/meta.yaml');
-    this.yaml.getJson().subscribe(data => {
-      this.YamlObject = data;
+  @ViewChildren(MatChip) chips!: QueryList<MatChip>;
+  matChipsArray: MatChip[] = [];
 
-      this.teamList = this.YamlObject['teams'];
-      this.teamGroups = this.YamlObject['teamGroups'];
-      this.teamVisible = [...this.teamList];
-    });
+  private LoadMaturityDataFromGeneratedYaml() {
     this.yaml.setURI('./assets/YAML/generated/generated.yaml');
-    // Function sets data
+
     this.yaml.getJson().subscribe(data => {
-      //console.log(this.radial_labels)
       this.YamlObject = data;
-      // console.log(this.YamlObject);
-
       var allDimensionNames = Object.keys(this.YamlObject);
-      // console.log(allDimensionNames);
-      for (var i = 0; i < allDimensionNames.length; i++) {
-        var allSubDimensionInThisDimension = Object.keys(
-          this.YamlObject[allDimensionNames[i]]
-        );
-        for (var j = 0; j < allSubDimensionInThisDimension.length; j++) {
-          this.segment_labels.push(allSubDimensionInThisDimension[j]);
-        }
-      }
-      // console.log(this.segment_labels);
+      var totalTeamsImplemented: number = 0;
+      var totalActivityTeams: number = 0;
+
+      this.AddSegmentLabels(allDimensionNames);
+
       for (var l = 0; l < this.maxLevelOfActivities; l++) {
-        var allDimensionNames = Object.keys(this.YamlObject);
-        for (var i = 0; i < allDimensionNames.length; i++) {
+        for (var d = 0; d < allDimensionNames.length; d++) {
           var allSubDimensionInThisDimension = Object.keys(
-            this.YamlObject[allDimensionNames[i]]
+            this.YamlObject[allDimensionNames[d]]
           );
-          for (var j = 0; j < allSubDimensionInThisDimension.length; j++) {
+          for (var s = 0; s < allSubDimensionInThisDimension.length; s++) {
             var allActivityInThisSubDimension = Object.keys(
-              this.YamlObject[allDimensionNames[i]][
-                allSubDimensionInThisDimension[j]
+              this.YamlObject[allDimensionNames[d]][
+                allSubDimensionInThisDimension[s]
               ]
             );
-            var tempData: cardSchema = {
-              Dimension: '',
-              SubDimension: '',
-              Level: '',
-              'Done%': -1,
-              Activity: [],
-            };
-            var totalTeamsImplemented: number = 0;
-            var totalActivityTeams: number = 0;
-            tempData['Dimension'] = allDimensionNames[i];
-            tempData['SubDimension'] = allSubDimensionInThisDimension[j];
-            tempData['Level'] = 'Level ' + (l + 1);
-            for (var k = 0; k < allActivityInThisSubDimension.length; k++) {
+            var level = 'Level ' + (l + 1);
+            var activity: activitySchema[] = [];
+            var activityCompletionStatus: number = -1;
+
+            for (var a = 0; a < allActivityInThisSubDimension.length; a++) {
```

Comment:
I personally would like to use a more speaking variable name like `activityIndex`. But I am ok with it as it is better than beforehand.

Reply:
I agree but the linter pushed it to the next line and made it really hard to read ... so it was a little bit of a balancing act.